duration.go

```diff
@@ -0,0 +1,27 @@
+package claircore
+
+import (
+	"errors"
+	"time"
+)
+
+// Duration is a serializeable [time.Duration].
+type Duration time.Duration
+
+// UnmarshalText implements [encoding.TextUnmarshaler].
+func (d *Duration) UnmarshalText(b []byte) error {
+	dur, err := time.ParseDuration(string(b))
+	if err != nil {
+		return err
+	}
+	*d = Duration(dur)
```

Comment:
I guess a `nil` check would make sense here, too, but I'm looking at https://cs.opensource.google/go/go/+/refs/tags/go1.24.2:src/time/time.go;l=1637 and even the standard library doesn't bother `nil` checking. Up to you